### PR TITLE
Update swc

### PR DESCRIPTION
**Pull request by kdy1**
This PR

 - updates swc crates to https://github.com/swc-project/swc/commit/224a2ea8b6b0ff92b68a17ce7655db471ab1f48c

 - resolves`https://github.com/vercel/next.js/discussions/30237?sort=new#discussioncomment-2759673`

 - resolves `https://github.com/vercel/next.js/discussions/30237?sort=new#discussioncomment-2780370`

image

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: b042ddf4ae34497586bef99269c165de34f6641e

`yarn testheadless test/development/acceptance/ReactRefreshLogBox-app-doc.test.ts` 
- ReactRefreshLogBox > _app syntax error shows logbox
- ReactRefreshLogBox > _document syntax error shows logbox

<details>
<summary>Expand output</summary>

  ● ReactRefreshLogBox › _app syntax error shows logbox

    expect(received).toMatchSnapshot()

    Snapshot name: `ReactRefreshLogBox _app syntax error shows logbox 1`

    - Snapshot  - 1
    + Received  + 2

    @@ -13,6 +13,7 @@
         :          ^^^^^^^^^
         `----

      Caused by:
          0: failed to process input file
    -     1: Syntax Error
    +     1: Parser config: Es(EsConfig { jsx: true, fn_bind: false, decorators: false, decorators_before_export: false, export_default_from: false, import_assertions: true, private_in_object: false, allow_super_outside_method: false })
    +     2: Syntax Error

      108 |     )
      109 |     expect(await session.hasRedbox(true)).toBe(true)
    > 110 |     expect(await session.getRedboxSource()).toMatchSnapshot()
          |                                             ^
      111 |
      112 |     await session.patch(
      113 |       'pages/_app.js',

      at Object.<anonymous> (development/acceptance/ReactRefreshLogBox-app-doc.test.ts:110:45)

  ● ReactRefreshLogBox › _document syntax error shows logbox

    expect(received).toMatchSnapshot()

    Snapshot name: `ReactRefreshLogBox _document syntax error shows logbox 1`

    - Snapshot  - 1
    + Received  + 2

    @@ -6,6 +6,7 @@
         :                                    ^
         `----

      Caused by:
          0: failed to process input file
    -     1: Syntax Error
    +     1: Parser config: Es(EsConfig { jsx: true, fn_bind: false, decorators: false, decorators_before_export: false, export_default_from: false, import_assertions: true, private_in_object: false, allow_super_outside_method: false })
    +     2: Syntax Error

      157 |     )
      158 |     expect(await session.hasRedbox(true)).toBe(true)
    > 159 |     expect(await session.getRedboxSource()).toMatchSnapshot()
          |                                             ^
      160 |
      161 |     await session.patch(
      162 |       'pages/_document.js',

      at Object.<anonymous> (development/acceptance/ReactRefreshLogBox-app-doc.test.ts:159:45)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| buildDuration | 16.3s | 16.4s | ⚠️ +95ms |
| buildDurationCached | 6.6s | 6s | -620ms |
| nodeModulesSize | 478 MB | 478 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.953 | 4.004 | ⚠️ +0.05 |
| / avg req/sec | 632.35 | 624.42 | ⚠️ -7.93 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.39 | 1.367 | -0.02 |
| /error-in-render avg req/sec | 1798.61 | 1828.43 | +29.82 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| 925.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42 kB | 42 kB | ✓ |
| main-HASH.js gzip | 28.8 kB | 28.8 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 72.4 kB | 72.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.36 kB | 1.36 kB | ✓ |
| _error-HASH.js gzip | 193 B | 193 B | ✓ |
| amp-HASH.js gzip | 308 B | 308 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 359 B | 359 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.72 kB | 5.72 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.65 kB | 2.65 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 391 B | 391 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16 kB | 16 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| buildDuration | 16.7s | 16.5s | -149ms |
| buildDurationCached | 5.9s | 6s | ⚠️ +60ms |
| nodeModulesSize | 478 MB | 478 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.609 | 3.577 | -0.03 |
| / avg req/sec | 692.64 | 698.82 | +6.18 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.26 | 1.282 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1984.39 | 1950.75 | ⚠️ -33.64 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| 925.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 42.7 kB | 42.7 kB | ✓ |
| main-HASH.js gzip | 29.2 kB | 29.2 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 73.5 kB | 73.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 324 B | 324 B | ✓ |
| dynamic-HASH.js gzip | 2.89 kB | 2.89 kB | ✓ |
| head-HASH.js gzip | 357 B | 357 B | ✓ |
| hooks-HASH.js gzip | 920 B | 920 B | ✓ |
| image-HASH.js gzip | 5.81 kB | 5.81 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 2.78 kB | 2.78 kB | ✓ |
| routerDirect..HASH.js gzip | 322 B | 322 B | ✓ |
| script-HASH.js gzip | 392 B | 392 B | ✓ |
| withRouter-HASH.js gzip | 317 B | 317 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 16.3 kB | 16.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 457 B | 457 B | ✓ |
| Overall change | 457 B | 457 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | kdy1/next.js swc-up-2 | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 546 B | 546 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: b042ddf4ae34497586bef99269c165de34f6641e

File: test/development/acceptance/__snapshots__/ReactRefreshLogBox-app-doc.test.ts.snap [hunk 1]
```diff
@@ -18,7 +18,8 @@ Error:
 
 Caused by:
     0: failed to process input file
-    1: Syntax Error"
+    1: Parser config: Es(EsConfig { jsx: true, fn_bind: false, decorators: false, decorators_before_export: false, export_default_from: false, import_assertions: true, private_in_object: false, allow_super_outside_method: false })
```
Comment (kdy1) [inline test/development/acceptance/__snapshots__/ReactRefreshLogBox-app-doc.test.ts.snap:21]: I added this to help users of `@swc/core`, but I can remove this for `next-swc` if required

**Review by Brooooooklyn — DISMISSED** [review]

**Review by Brooooooklyn — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Comment by PupoSDC** [comment]
@kdy1 Just wanted to confirm that the newest canary indeed fixed all my issues. Thanks a lot! 🤗

**Pull request merged**